src/Raven.Server/Documents/Document.cs

```diff
@@ -7,7 +7,7 @@
 
 namespace Raven.Server.Documents
 {
-    public class Document
+    public class Document : IDisposable
```

Comment:
This would require to change how we deal with `Document` all over the places where we use that object because we needed to dispose it everywhere. We typically just dispose `BlittableJsonReaderObject`, `LazyStringValue` explicitly. So instead of introducing `IDisposable` here we can dispose what's needed inside the dispose method of `PatchResult`, right?

Reply:
What can be the problem? that I will dispose an external value that we still need?

Reply:
Question: 
I know that we cache `LazyStringValue`s inside the context.
Might be that the `Id`/`LowerId` are cached and we are disposing them?

Reply:
IIRC the JsonOperationContext contains fieldName cache, and this is not related here, because the Document.Id and LowerId comes from allocator. I might be wrong... @ayende do you remember?

Reply:
I and @karmeli87 followed it and the pointer for the `LazyString` comes from the table itself (Memory Map File) and in this situation the `LazyString` object will not dispose the memory. It will dispose in a situation when the memory was  allocated specially for the `LazyString` but it is safe to dispose in this situation too. There is a reuse of the `LazyString` object but the context will renew the `LasyString` object before so no problem to dispose the `LasyString` object. The `FieldName` cache related to `DocumentBuilder` while it building a document.

Reply:
If you read from the mmap blittable, that gives you a lazy string instance on the mmap, nothing is allocated in this case.